### PR TITLE
ROS2 compatibility

### PR DESCRIPTION
**Pull request by Timple**
This allows ros ROS1 and ROS2 compilation. No need to keep branches in sync.

### CODE REVIEW

**Comment by tilk** [comment]
Thanks! Can't test it right now, as I'm on holiday, but looks fine.

**Comment by Timple** [comment]
Have a great holliday!

When you get back, would you mind bumping the release (and tag it?).

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,16 +1,38 @@
...
+  catkin_package(CATKIN_DEPENDS message_runtime std_msgs)
+
+elseif(${ROS_VERSION} EQUAL 2)
+  find_package(ament_cmake REQUIRED)
+  find_package(builtin_interfaces REQUIRED)
+  find_package(rosidl_default_generators REQUIRED)
+
+  rosidl_generate_interfaces(${PROJECT_NAME} ADD_LINTER_TESTS
+    msg/Message.msg
+  )
```
Comment (samiamlabs) [inline CMakeLists.txt:33]: Looks like you missed adding the dependency for std_msgs here. It won't build for me on ROS2 (humble) without it.
I think it should be:
```cmake
rosidl_generate_interfaces(${PROJECT_NAME} ADD_LINTER_TESTS
    msg/Message.msg
    DEPENDENCIES std_msgs
  )

```

**Comment by tilk** [comment]
I'll return on Monday and then I'll take a look.

File: CMakeLists.txt (hunk 1 above)
Comment (Timple) [reply CMakeLists.txt:33]: Looks like you're right. Although it compiled just fine on my PC...
I'll add anyway to be specific.

Comment (samiamlabs) [reply CMakeLists.txt:33]: Yes, odd that it worked for you and not me... I tried building several times etc, but could not get it to work without explicit dependencies. 

Tested it on humble in a docker container with Ubuntu 22.04 LTS. Not sure if that matters.

**Pull request merged**